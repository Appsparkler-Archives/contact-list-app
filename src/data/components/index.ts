export * from "./standard-select-field";
