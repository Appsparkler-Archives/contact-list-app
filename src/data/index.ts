export * from "./contacts";
export * from "./components";
