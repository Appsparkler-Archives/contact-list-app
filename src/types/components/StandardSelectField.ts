import { SelectChangeEvent } from "@mui/material";
import React from "react";

export interface IStandSelectFieldOptionItem {
  name: string | number;
  value: string | number;
}

export type TStandardSelectFieldFC = React.FC<{
  options: IStandSelectFieldOptionItem[];
  selectedOption: string | number;
  label: string;
  onChange: (event: SelectChangeEvent<any>, child: React.ReactNode) => void;
}>;
