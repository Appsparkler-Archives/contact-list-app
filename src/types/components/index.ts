export * from "./StandardSelectField";
export * from "./FilterChip";
export * from "./CreateContactForm";
export * from "./ContactListAccordion";
export * from "./ContactForm";
export * from "./DeleteContact";
export * from "./FullScreenModalTransition";
export * from "./ViewContactModal";
