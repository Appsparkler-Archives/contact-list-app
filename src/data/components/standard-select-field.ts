import { IStandSelectFieldOptionItem } from "types/components/StandardSelectField";

export const standardSelectFieldOptions: IStandSelectFieldOptionItem[] = [
  {
    name: "Female",
    value: "female",
  },
  {
    name: "Male",
    value: "male",
  },
];
