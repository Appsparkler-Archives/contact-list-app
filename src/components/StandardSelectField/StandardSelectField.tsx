import { TSelectChangeEventHandler, TStandardSelectFieldFC } from "types";
import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import Select from "@mui/material/Select";
import { useCallback } from "react";

export const StandardSelectField: TStandardSelectFieldFC = ({
  selectedOption,
  onChange,
  options,
  label,
  name,
  id,
}) => {
  const handleChange = useCallback<TSelectChangeEventHandler>(
    ({ target: { value, name } }) => {
      onChange(name, value);
    },
    [onChange]
  );
  return (
    <FormControl variant="standard" fullWidth>
      <InputLabel required id={id}>
        {label}
      </InputLabel>
      <Select<string>
        labelId={id}
        onChange={handleChange}
        label={label}
        required
        name={name}
        defaultValue={selectedOption}
      >
        {options.map((option) => (
          <MenuItem key={option.value} value={option.value}>
            {option.name}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};
