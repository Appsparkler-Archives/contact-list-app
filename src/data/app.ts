import { IContactFormData } from "types";

export type TAppFC = React.FC<{
  contacts: IContactFormData[];
}>;
