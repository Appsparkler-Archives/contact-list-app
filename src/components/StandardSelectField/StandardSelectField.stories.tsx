import type { Meta, StoryObj } from "@storybook/react";
import { StandardSelectField } from "./StandardSelectField";
import { standardSelectFieldOptions } from "data";

const meta = {
  title: "Components/Standard Select Field",
  component: StandardSelectField,
  tags: ["components"],
} satisfies Meta<typeof StandardSelectField>;

export default meta;

type Story = StoryObj<typeof meta>;

export const standardSelectField: Story = {
  args: {
    id: "standard-select-field-id",
    label: "Gender",
    options: standardSelectFieldOptions,
    selectedOption: standardSelectFieldOptions[0].value,
  },
};
