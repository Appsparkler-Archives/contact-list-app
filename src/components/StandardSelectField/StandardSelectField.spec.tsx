import renderer from "react-test-renderer";
import { TStandardSelectFieldFC } from "types/components/StandardSelectField";
import InputLabel from "@mui/material/InputLabel";
import MenuItem from "@mui/material/MenuItem";
import FormControl from "@mui/material/FormControl";
import Select from "@mui/material/Select";
import { standardSelectFieldOptions } from "data";
import { noop } from "lodash/fp";

const SelectField: TStandardSelectFieldFC = ({
  selectedOption,
  onChange,
  options,
  label,
}) => {
  return (
    <FormControl variant="standard" sx={{ m: 1, minWidth: 120 }}>
      <InputLabel id="demo-simple-select-standard-label">{label}</InputLabel>
      <Select
        labelId="demo-simple-select-standard-label"
        value={selectedOption}
        onChange={onChange}
        label="Age"
        required
      >
        {options.map((option) => (
          <MenuItem key={option.value} value={option.value}>
            <em>{option.name}</em>
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};

describe("SelectField FC", () => {
  it("should match snapshot", async () => {
    const component = renderer.create(
      <SelectField
        options={standardSelectFieldOptions}
        selectedOption={standardSelectFieldOptions[0].value}
        onChange={noop}
        label={"Gender"}
      />
    );
    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
