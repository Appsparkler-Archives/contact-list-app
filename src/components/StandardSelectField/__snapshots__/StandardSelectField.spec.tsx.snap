// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField FC should match snapshot 1`] = `
<div
  className="MuiFormControl-root css-1869usk-MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={true}
    id="demo-simple-select-standard-label"
  >
    Gender
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl  css-m5hdmq-MuiInputBase-root-MuiInput-root-MuiSelect-root"
    onClick={[Function]}
    required={true}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="demo-simple-select-standard-label"
      className="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <em>
        Female
      </em>
    </div>
    <input
      aria-hidden={true}
      aria-invalid={false}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={true}
      tabIndex={-1}
      value="female"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
